test/libdl.jl

```diff
@@ -171,7 +171,7 @@ let dl = C_NULL
     end
 end
 
-if OS_NAME in (:Linux, :FreeBSD)
+if Sys.KERNEL in (:Linux, :FreeBSD)
```

Comment:
This is the list we define the tested function on. I assume it's whatever platforms where soname is a thing.